internal/compiler/Unit.go

```diff
@@ -128,3 +142,16 @@ func (u *Unit) Scan() bool {
 	}
 	return !u.HasErrors()
 }
+
+func (u *Unit) Parse() bool {
+	if u.compilationStage == CompilationStageScanned {
+		if u.rootFile.Scan() {
```

Comment:
I think this should be Parse();
```suggestion
		if u.rootFile.Parse() {
```